### PR TITLE
New upload dialog

### PR DESCRIPTION
**Pull request by mbernasocchi**
this PR adds the possibility of lettin the user choose what it would like to upload

### CODE REVIEW

**Comment by ptormene** [comment]
I've fixed the TODO

**Comment by mbernasocchi** [comment]
Not wip any more. Please do a cr

File: svir.py [hunk 1]
```diff
@@ -1055,8 +1060,8 @@ def upload(self):
             file_size_mb, self.iface, project_definition)
         if dlg.exec_():
             project_definition['title'] = dlg.ui.title_le.text()
-            zone_label_field = dlg.ui.zone_label_field_cbx.currentText()
-            project_definition['zone_label_field'] = zone_label_field
```
Comment (ptormene) [inline svir.py]: Why did you remove the selection of the zone_label_field? It is needed by the web viewer, to know which field contains the zone labels.

File: upload_settings_dialog.py [hunk 2]
```diff
@@ -62,23 +62,16 @@ def __init__(self, upload_size, iface, project_definition):
...
-        self.zone_label_field_is_specified = False
-        reload_attrib_cbx(
-            self.ui.zone_label_field_cbx, iface.activeLayer(), True)
-        # pre-select the field if it's specified in the project definition
-        if 'zone_label_field' in self.project_definition:
-            zone_label_idx = self.ui.zone_label_field_cbx.findText(
-                self.project_definition['zone_label_field'])
-            if zone_label_idx != -1:
-                self.ui.zone_label_field_cbx.setCurrentIndex(zone_label_idx)
-                self.zone_label_field_is_specified = True
```
Comment (ptormene) [inline upload_settings_dialog.py]: Same as above. We need to keep the selection of the zone_label_field, pre-populating the combobox if the project definition already contains such information

File: ui/ui_upload_settings.ui [hunk 3]
```diff
@@ -7,27 +7,55 @@
...
+   <string>Upload</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
-    <widget class="QLabel" name="head_msg_lbl">
+    <widget class="QLabel" name="situation_lbl">
      <property name="text">
-      <string>The active layer will be uploaded to the Openquake Platform</string>
+      <string>It looks like this layer was downloaded from a platform. 
+You can create a new layer or just upload the last project definition</string>
```
Comment (ptormene) [inline ui/ui_upload_settings.ui]: Please change `last -> current`

File: ui/ui_upload_settings.ui [hunk 4]
```diff
@@ -47,14 +75,14 @@
       <widget class="QLineEdit" name="title_le"/>
      </item>
      <item row="1" column="0">
-      <widget class="QLabel" name="label">
+      <widget class="QLabel" name="description_lbl">
        <property name="text">
-        <string>Zone labels field</string>
```
Comment (ptormene) [inline ui/ui_upload_settings.ui]: Same as above. We need both the zone label field and the description

File: upload_settings_dialog.py [hunk 5]
```diff
@@ -88,44 +81,51 @@ def __init__(self, upload_size, iface, project_definition):
...
+            title_lbl = tr('New layer title')
+            description_lbl = tr('New layer description')
+
+        self.ui.title_lbl.setText(title_lbl)
+        self.ui.description_lbl.setText(description_lbl)
+        self.ui.explaination_lbl.setText(explaination_lbl)
 
     def set_ok_button(self):
         self.ok_button.setEnabled(
-            self.zone_label_field_is_specified
```
Comment (ptormene) [inline upload_settings_dialog.py]: Same as above

File: upload_settings_dialog.py [hunk 6]
```diff
@@ -88,44 +81,51 @@ def __init__(self, upload_size, iface, project_definition):
...
-            and self.ui.confirm_chk.isChecked())
+            self.ui.title_le.text() and self.ui.confirm_chk.isChecked())
 
-    @pyqtSlot(str)
-    def on_zone_label_field_cbx_currentIndexChanged(self):
-        zone_label_field = self.ui.zone_label_field_cbx.currentText()
-        self.zone_label_field_is_specified = (zone_label_field != '')
-        self.set_ok_button()
```
Comment (ptormene) [inline upload_settings_dialog.py]: Same as above

**Comment by ptormene** [comment]
When we upload, we should save the layer id into the project definition, so we will be able to update the project afterwards. This looks simple: we should build the name as something similar to `oqplatform:file_stem` (including the path without '/' and probably replacing '-' with '_')

File: upload_settings_dialog.py [hunk 7]
```diff
@@ -88,44 +81,51 @@ def __init__(self, upload_size, iface, project_definition):
...
-        if self.ui.update_chk.isChecked():
-            self.ui.head_msg_lbl.setText(self.head_msg_update)
-        else:
-            self.ui.head_msg_lbl.setText(self.head_msg_new)
+            explaination_lbl = tr(
+                'A new layer will be created on the OpenQuake Platform.'
+                '\n(About %s MB of data will be transmitted)'
+                % self.upload_size)
+            title_lbl = tr('New layer title')
+            description_lbl = tr('New layer description')
```
Comment (ptormene) [inline upload_settings_dialog.py]: I would change this into `New layer abstract` (because it will be written into the 'Abstract' metadata field)

File: svir.py (hunk 1 above)
Comment (mbernasocchi) [inline svir.py]: Because we discussed it should be implemented in the project manager

**Comment by ptormene** [comment]
LGTM, and waiting for @mbernasocchi for his confirmation before merging (Marco, feel free to merge, if it LGTY too)

**Comment by mbernasocchi** [comment]
LGTM
